exporters/metrics/src/main/java/com/google/cloud/opentelemetry/metric/ResourceTranslator.java

```diff
@@ -0,0 +1,122 @@
+        if (value != null) {
+          builder.putLabels(getLabelName(), value.toString());
+          return;
+        }
+      }
+      fallbackLiteral().ifPresent(value -> builder.putLabels(getLabelName(), value));
+    }
+
+    public static AttributeMapping create(String labelName, AttributeKey<?> otelKey) {
+      return new AutoValue_ResourceTranslator_AttributeMapping(
+          labelName, java.util.Collections.singletonList(otelKey), Optional.empty());
+    }
+
+    public static AttributeMapping create(
+        String labelName, java.util.List<AttributeKey<?>> otelKeys) {
+      return new AutoValue_ResourceTranslator_AttributeMapping(
+          labelName, otelKeys, Optional.empty());
+    }
+
+    public static AttributeMapping create(
+        String labelName, java.util.List<AttributeKey<?>> otelKeys, String fallbackLiteral) {
+      return new AutoValue_ResourceTranslator_AttributeMapping(
+          labelName, otelKeys, Optional.of(fallbackLiteral));
+    }
+  }
+
+  private static List<AttributeMapping> GCE_INSTANCE_LABELS =
+      java.util.Arrays.asList(
+          AttributeMapping.create("zone", ResourceAttributes.CLOUD_AVAILABILITY_ZONE),
+          AttributeMapping.create("instance_id", ResourceAttributes.HOST_ID));
+  private static List<AttributeMapping> K8S_POD_LABELS =
+      java.util.Arrays.asList(
+          AttributeMapping.create("location", ResourceAttributes.CLOUD_AVAILABILITY_ZONE),
+          AttributeMapping.create("cluster_name", ResourceAttributes.K8S_CLUSTER_NAME),
+          AttributeMapping.create("namespace_name", ResourceAttributes.K8S_NAMESPACE_NAME),
+          AttributeMapping.create("pod_name", ResourceAttributes.K8S_POD_NAME));
+  private static List<AttributeMapping> AWS_EC2_INSTANCE_LABELS =
+      java.util.Arrays.asList(
+          AttributeMapping.create("instance_id", ResourceAttributes.HOST_ID),
+          AttributeMapping.create("region", ResourceAttributes.CLOUD_AVAILABILITY_ZONE),
+          AttributeMapping.create("aws_account", ResourceAttributes.CLOUD_ACCOUNT_ID));
+  private static List<AttributeMapping> GENERIC_TASK_LABELS =
+      java.util.Arrays.asList(
+          AttributeMapping.create(
+              "location",
+              java.util.Arrays.asList(
+                  ResourceAttributes.CLOUD_AVAILABILITY_ZONE, ResourceAttributes.CLOUD_REGION),
+              "global"),
+          AttributeMapping.create("namespace", ResourceAttributes.SERVICE_NAMESPACE),
+          AttributeMapping.create("job", ResourceAttributes.SERVICE_NAME),
+          AttributeMapping.create("task_id", ResourceAttributes.SERVICE_INSTANCE_ID));
+
+  /** Converts a Java OpenTelemetyr SDK resoruce into a MonitoredResource from GCP. */
+  public static MonitoredResource mapResource(Resource resource) {
+    String platform = resource.getAttribute(ResourceAttributes.CLOUD_PLATFORM);
+    if (platform == null) {
+      return mapBase(resource, "generic_task", GENERIC_TASK_LABELS);
+    }
```

Comment:
Do we need both this and the `default` case on 108?

Reply:
Yeah. https://docs.oracle.com/javase/specs/jls/se8/html/jls-14.html#jls-14.11

You can't switch on `null` in Java, so you have to filter it out ahead of time or get NPE